apps/api-extractor/src/collector/Collector.ts

```diff
@@ -425,50 +424,54 @@ export class Collector {
       this._collectReferenceDirectives(astEntity);
     }
 
-    if (exportedName) {
-      entity.addExportName(exportedName);
+    if (exportName) {
+      if (parent) {
+        entity.addLocalExportName(exportName, parent);
+      } else {
+        entity.addExportName(exportName);
+      }
     }
 
     return entity;
   }
 
-  private _createEntityForIndirectReferences(
-    astEntity: AstEntity,
-    alreadySeenAstEntities: Set<AstEntity>
-  ): void {
-    if (alreadySeenAstEntities.has(astEntity)) {
-      return;
-    }
+  private _recursivelyCreateEntities(astEntity: AstEntity, alreadySeenAstEntities: Set<AstEntity>): void {
```

Comment:
I renamed this method to `_recursivelyCreateEntities` because the previous name was inaccurate. The previous name implied that we were only creating entities for indirect references, when in fact we also create entities for things within `AstNamespaceImport` entities.